### PR TITLE
Added support for OkHttp

### LINKED ISSUE #204: Add support for OkHttp
See https://github.com/square/okhttp/wiki/Interceptors

Inspired by #203

### PR DESCRIPTION
**Pull request by whiskeysierra**
Fixes #204

### CODE REVIEW

File: logbook-okhttp/src/main/java/org/zalando/logbook/okhttp/RemoteResponse.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    @Override
+    public int getStatus() {
+        return response.code();
+    }
+
+    @Override
+    public String getProtocolVersion() {
+        // TODO find the real thing
+        return "HTTP/1.1";
```
Comment (whiskeysierra) [inline logbook-okhttp/src/main/java/org/zalando/logbook/okhttp/RemoteResponse.java]: `return response.protocol().toString();` (+ to upper case?)

https://github.com/square/okhttp/blob/f23067f6ccba94db598485530d722fcb60f9e3e0/okhttp/src/main/java/okhttp3/Protocol.java#L41

File: logbook-okhttp/src/main/java/org/zalando/logbook/okhttp/LocalRequest.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    @Override
+    public String getHost() {
+        return request.url().host();
+    }
+
+    @Override
+    public Optional<Integer> getPort() {
+        // TODO empty for 80/443?
```
Comment (whiskeysierra) [inline logbook-okhttp/src/main/java/org/zalando/logbook/okhttp/LocalRequest.java]: How are we dealing with this in `logbook-httpclient`?

**Comment by whiskeysierra** [comment]
@lukasniemeier-zalando @AlexanderYastrebov Can you take a look?

File: logbook-okhttp/pom.xml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            <artifactId>logbook-core</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-nop</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.github.rest-driver</groupId>
+            <artifactId>rest-client-driver</artifactId>
```
Comment (AlexanderYastrebov) [inline logbook-okhttp/pom.xml]: `<scope>test</scope>`

File: logbook-okhttp/src/main/java/org/zalando/logbook/okhttp/LogbookInterceptor.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    @Override
+    public Response intercept(final Chain chain) throws IOException {
+        final LocalRequest request = new LocalRequest(chain.request());
+        final Optional<Correlator> correlator = logbook.write(request);
+
+        final RemoteResponse response = new RemoteResponse(chain.proceed(request.toRequest()));
+
+        if (correlator.isPresent()) {
```
Comment (AlexanderYastrebov) [inline logbook-okhttp/src/main/java/org/zalando/logbook/okhttp/LogbookInterceptor.java]: `ifPresent`?

File: logbook-okhttp/pom.xml (hunk 3 above)
Comment (whiskeysierra) [reply logbook-okhttp/pom.xml]: Scope is specified in the parent pom, so I don't have to repeat myself all the time.

File: logbook-okhttp/src/main/java/org/zalando/logbook/okhttp/LogbookInterceptor.java (hunk 4 above)
Comment (whiskeysierra) [reply logbook-okhttp/src/main/java/org/zalando/logbook/okhttp/LogbookInterceptor.java]: `write` throws an `IOException` and the functional interfaces don't allow checked exceptions.

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**